### PR TITLE
Updated fetchSingle to use new Reader

### PR DESCRIPTION
**Pull request by obany**
Updated fetchSingle to use the new Reader class the same as fetch. 

The old fetchSingle method was very unstable and failed to decode the messages without any sensible feedback as to what the failure was.

### CODE REVIEW

**Comment by sobolev-alexey** [comment]
The updated fetchSingle function duplicates about 90% of the `fetch` function source code.
It might be good idea to refactor both functions or to make one function to call another internally.

**Comment by obany** [comment]
Added a limit param to fetch, this is also useful at other times if you don't want to wait for all new messages to be returned.

The fetchSingle method utilises this new param to just get 1 item and so folds all the complex work into one method.

File: src/index.js [hunk 1]
```diff
@@ -148,35 +149,20 @@ const fetch = async (root, selectedMode, sidekey, callback) => {
...
-            // Unmask the message
-            const { payload, next_root } = decode(message, sidekey, root)
-            // Return payload
-            return { payload, nextRoot: next_root }
-        } catch (e) {
-            console.error('failed to parse: ', e)
-        }
-    }
+const fetchSingle = async (root, selectedMode, sidekey) => {
+    const response = await fetch(root, selectedMode, sidekey, undefined, 1)
```
Comment (sobolev-alexey) [inline src/index.js:161]: Is there a specific reason to not explicitly set the `callback` to `null`? 

same is on line 163.

Comment (obany) [reply src/index.js:161]: Not sure I understand your question, we don't have a callback passed to fetchSingle we just return the value, so we pass undefined to fetch to not use the callback mechanism.

**Review by sobolev-alexey — APPROVED** [review]

**Pull request merged**